src/AppShed/Extensions/StorageBundle/Controller/ApiController.php

```diff
@@ -0,0 +1,444 @@
+     * @Method({"GET", "POST"})
+     * @return array
+     * @Template()
+     */
+    public function editAction(Request $request, Api $api, $uuid)
+    {
+        $app = $this->getApp($request);
+        $appParams = $this->getExtensionAppParameters($request);
+        if ($api->getApp() != $app) {
+            throw new NotFoundHttpException("Api with uuid $uuid not found");
+        }
+
+        $form = $this->createForm(new ApiEditType($api), $api, [
+            'action' => $this->generateUrl('api_edit', array_merge($appParams, ['uuid' => $api->getUuid()])),
+            'method' => 'POST'
+        ]);
+
+        switch ($api->getAction()) {
+            case Api::ACTION_UPDATE:
+            case Api::ACTION_DELETE: {
+                $form
+                    ->remove('fields')
+                    ->remove('groupField')
+                    ->remove('orderField')
+                    ->remove('orderDirection');
+            } break;
+            case Api::ACTION_INSERT: {
+                $form
+                    ->remove('fields')
+                    ->remove('filters')
+                    ->remove('groupField')
+                    ->remove('orderField')
+                    ->remove('orderDirection')
+                    ->remove('limit');
+            } break;
+        }
+
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($api);
+            $em->flush();
+            $this->get('session')->getFlashBag()->add(
+                'success',
+                'Your changes were saved'
+            );
+            return $this->redirect($this->generateUrl('api_list', $appParams));
+        }
+
+        $data = [
+            'api' => $api,
+            'form'   => $form->createView(),
+            'appParams' => $appParams
+        ];
+        return $data;
+    }
+
+    /**
+     * @Route("/api/{uuid}/delete", name="api_delete")
+     * @param Api $api
+     * @param $uuid
+     * @ParamConverter("api", class="AppShed\Extensions\StorageBundle\Entity\Api", options={"uuid"="uuid"})
+     * @Method({"GET", "POST"})
+     * @return array
+     */
+    public function deleteAction(Request $request, Api $api, $uuid)
+    {
+        $app = $this->getApp($request);
+        $appParams = $this->getExtensionAppParameters($request);
+        if ($api->getApp() != $app) {
+            throw new NotFoundHttpException("Api with uuid $uuid not found");
+        }
+
+        $em = $this->getDoctrine()->getManager();
+        $em->remove($api);
+        $em->flush();
+
+        return $this->redirect($this->generateUrl('api_list', $appParams));
+    }
+
+    /**
+     * @Route("/api/{uuid}", name="api_show")
+     * @param Api $api
+     * @ParamConverter("api", class="AppShed\Extensions\StorageBundle\Entity\Api", options={"uuid"="uuid"})
+     * @Method({"GET", "POST"})
```

Comment:
This action needs to be broken into two actions, one for `GET` that will only support `ACTION_SELECT`
Another for `POST` that will do all the actions